src/core/project/qgsproject.cpp

```diff
@@ -2668,6 +2668,10 @@ void QgsProject::onMapLayersAdded( const QList<QgsMapLayer *> &layers )
     if ( ! layer->isValid() )
       return;
 
+    if ( QgsVectorLayer *vlayer = qobject_cast<QgsVectorLayer *>( layer ) )
+      if ( vlayer->dataProvider() )
+        vlayer->dataProvider()->setProviderProperty( QgsVectorDataProvider::EvaluateDefaultValues, ( bool )( mFlags & Qgis::ProjectFlag::EvaluateDefaultValuesOnProviderSide ) );
```

Comment:
I guess this is not the only [flag](https://github.com/qgis/QGIS/blob/master/src/core/project/qgsproject.cpp#L536) for which you will get an issue

Reply:
Added `Qgis::ProjectFlag::TrustStoredLayerStatistics` flag